### ISSUE TITLE
Error while writing the code in python functions

### ISSUE THREAD

**Issue by hiteshpara** [mljar/mercury#16]
Hi,
My Jupiter notebook code looks like follows:
```
---
title: My first notebook
description: my first notebook on mercury
show-code: False
params:
    a:
        input: slider
        label: This is slider label
        value: 5
        min: 0
        max: 100
    b:
        label: This is numeric label
        input: numeric
        value: 5.5
        min: 0
        max: 100
        step: 1
---
a=5
b=6
def add(a,b):
    return a+b
print(add(a+b))
```
while running it in the web app it gives an error add is not defined.
can any one help me with how to overcome this?

**Comment by pplonski** [comment]
Do you have

```
a=5
b=6
```

as a separate cell?

**Comment by hiteshpara** [comment]
No,
In same cell.
cell No 1:
---
title: My first notebook
description: my first notebook on mercury
show-code: False
params:
    a:
        input: slider
        label: This is slider label
        value: 5
        min: 0
        max: 100
    b:
        label: This is numeric label
        input: numeric
        value: 5.5
        min: 0
        max: 100
        step: 1
---
cell no 2:
a=5
b=6
def add(a,b):
    return a+b

cell no 3:
print(add(a+b))

**Comment by pplonski** [comment]
Two things.
1. Please set variables in the separate cell. Please move function definition into other cell.
2. Please try in your code: `print(add(a, b))` (currently you have + sign instead comma).

**Comment by pplonski** [comment]
I've reporduced your notebook after changes:

![image](https://user-images.githubusercontent.com/6959032/150484439-70c4ec5b-786e-4e67-b80b-f180a2c775c7.png)

The application (in watch mode):

![image](https://user-images.githubusercontent.com/6959032/150484519-f2ea875b-8243-40f7-bba0-1d6d1f573760.png)

**Comment by hiteshpara** [comment]
Thank You for your help.
It worked when I placed the variable in a different cell.
One more question:
is it possible to give text as input in the web application?
for suppose:
I have a Keras model and load it in notebook;
code: model=keras.load(abc.h5)
text="i am unhappy"
pred=model.pred(text)
print(pred)
is this possible in mercury?
and is it possible give random text in web app to test the model instead of
giving it in YAML?

**Comment by pplonski** [comment]
@hiteshpara great that it works for you!

I will add text input soon (should be ready for next week). I've already implemented a file input. There will be a new version release in the next week.

I don't  understand the last question, what do you mean by giving random text in web app? You will be able to give any string with text input. It will be possible to set a default value in text input. Please, could you explain the last question?

**Comment by hiteshpara** [comment]
Sure. eagerly waiting for the next version.
for suppose:
I am a data scientist and I developed one ML model to identify the given
text is a positive, negative, or neutral sentiment.
so now I want to integrate this with mercury-like this:
I give the mercury web app to my client and my client wants to give
inputs(according to his needs and we don't know what the client will give)
to the model in the web app instead of in the code.
my question is how can we do this in mercury.
I think in the later version it is possible.
and gave me any suggestion that how this mercury will help data
scientists in the particle world.

**Comment by pplonski** [comment]
@hiteshpara I'm closing the issue. I will let you know when text input is ready.

**Comment by hiteshpara** [comment]
Hi Piotr,
Yes, it is Working.
Thank You.
and after done with the experiment I want to delete my notebook. is it
possible?

**Comment by hiteshpara** [comment]
and also can you provide some sample code for file input?
That would be really helpful.
Thanks and regards
Para

**Comment by pplonski** [comment]
Hi @hiteshpara,

1. How to delete a notebook?

I've added issue to add a `delete` command https://github.com/mljar/mercury/issues/18. 

Right now if you want to remove notebook you need to login to Admin Panel and remove the notebook there. How to login to Admin Panel? First you need to create a super user:

```
mercury createsuperuser
```

(it will ask for username, password and email - email can ba fake, it is not used)

Then go to the URL:

```
https://127.0.0.1/admin
```

You will have there a Notebooks, click on them and find in the list notebook that you want to remove. Select it and choose action from the top menu.

2. How to use file as input?

The description of file field is here https://github.com/mljar/mercury#file so you need to define in YAML:

```yaml
params:
    filename:
        label: This is file label
        input: file
        maxFileSize: 1MB
```

Then in the notebook you need to have a variable `filename` 

In the first cell:
```python
filename = "input_file_name.csv"
```

In some other cell, just use the variable as a filename:

```python
import pandas as pd
df = pd.read_csv(filename)
```

Sorry that the detail guide is not ready - it is a new feature :)

Please let me know if you manage to use it successfully.

**Comment by hiteshpara** [comment]
Hi Pioter,
Thanks for the help.
Right now if i run mercury runserver --runworker it is opening in
http://127.0.0.1:8000/ which means it is using 8000 port.
is it possible to change that port? For example in my env 8000 port is
occupied by some other application so i want to this mercury to be
opened in another port,

**Comment by pplonski** [comment]
Please try for example

```
mercury runserver 8001 --runworker
```

**Comment by hiteshpara** [comment]
It's working.
Thank you.
So we can give any port during the runtime?
